spec/utils.ts

```diff
@@ -165,23 +168,47 @@ export function mockIsThirdPartyManager(
 }
 
 /**
- * Mocks the "itemExists" method of the thirdPartyAPI module.
- * This mock requires the thirdPartyAPI's itemExists method to be mocked first.
+ * Mocks the "exists" method of the ItemCuration Model
+ * This mock requires the ItemCuration's exists method to be mocked first.
+ *
+ * @param itemId - The id of the item that will be checked for a curation.
+ * @param exists - If the item with the given URN exists or not.
+ */
+export function mockThirdPartyItemCurationExists(
+  itemId: string,
+  exists: boolean
+) {
+  if (!(ItemCuration.exists as jest.Mock).mock) {
+    throw new Error(
+      "ItemCuration.exists should be mocked to use mockThirdPartyItemExists but isn't"
+    )
+  }
+
+  ;(ItemCuration.exists as jest.MockedFunction<
+    typeof ItemCuration.exists
+  >).mockImplementationOnce((idToCheck) =>
+    Promise.resolve(idToCheck === itemId && exists)
+  )
+}
+
+/**
+ * Mocks the "exists" method of the ItemCuration Model
+ * This mock requires the ItemCuration's exists method to be mocked first.
  *
  * @param urn - The URN of the item that will be checked for existence.
  * @param exists - If the item with the given URN exists or not.
  */
-export function mockThirdPartyItemExists(urn: string, exists: boolean) {
-  if (!(thirdPartyAPI.itemExists as jest.Mock).mock) {
+export function mockThirdPartyURNExists(urn: string, exists: boolean) {
+  if (!(peerAPI.fetchWearables as jest.Mock).mock) {
     throw new Error(
-      "exists should be mocked to mock the withModelExists middleware but it isn't"
+      "peerAPI.fetchWearables should be mocked to use mockThirdPartyURNExists but isn't"
     )
   }
 
-  ;(thirdPartyAPI.itemExists as jest.MockedFunction<
-    typeof thirdPartyAPI.itemExists
-  >).mockImplementationOnce((URNToCheck) =>
-    Promise.resolve(URNToCheck === urn && exists)
+  ;(peerAPI.fetchWearables as jest.MockedFunction<
+    typeof peerAPI.fetchWearables
+  >).mockImplementationOnce(([urnToCheck]) =>
+    Promise.resolve(urnToCheck === urn && exists ? [tpWearableMock] : [])
```

Comment:
What if, as we're going to return a mock, we do it by returning something received as a parameter? I mean, we often change the items and such in the tests and this could be a bit of help.

Reply:
I'm fine with that!